### ISSUE TITLE
repo manager fails to add the reference set for 1000 Genomes, hs37d5.fa.gz

### ISSUE THREAD

**Issue by kozbo** [ga4gh/server#1049]
While trying to add some parts of the 1000 Genome dat set to my Azure vm I found that I got an error while trying to add the reference fa.gz file.

The import sort of fails and sort of succeeds. Here are the details:
```
(ga4gh-server-env)azureuser@kobo-vm:/srv/ga4gh/ko-server$ ga4gh_repo add-referenceset /srv/ga4gh/ga4gh-example-data /mnt/1000Genome/reference/hs37d5.fa.gz
[fai_load] failed to load .gzi index: /srv/ga4gh/ga4gh-example-data/referenceSets/hs37d5/hs37d5.fa.gz[.gzi]
Traceback (most recent call last):
  File "/srv/ga4gh/ga4gh-server-env/bin/ga4gh_repo", line 9, in <module>
    load_entry_point('ga4gh==0.2.0a2.dev170+ngcce89bc', 'console_scripts', 'ga4gh_repo')()
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/cli.py", line 1672, in repo_main
    runner.run()
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/cli.py", line 1445, in run
    self.filePath, self.moveMode, self.metadata)
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/repo_manager.py", line 373, in addReferenceSet
    md5checksum = getReferenceChecksum(fileDestPath)
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/repo_manager.py", line 27, in getReferenceChecksum
    inputFile = pysam.FastaFile(fastaFile)
  File "pysam/cfaidx.pyx", line 113, in pysam.cfaidx.FastaFile.__cinit__ (pysam/cfaidx.c:2025)
  File "pysam/cfaidx.pyx", line 149, in pysam.cfaidx.FastaFile._open (pysam/cfaidx.c:2845)
IOError: could not open file `/srv/ga4gh/ga4gh-example-data/referenceSets/hs37d5/hs37d5.fa.gz`
```
But the file is actually there (though a symbolic link), and an index has been created
```
(ga4gh-server-env)azureuser@kobo-vm:/srv/ga4gh/ko-server$ ls -l /srv/ga4gh/ga4gh-example-data/referenceSets/hs37d5
total 4
lrwxrwxrwx 1 azureuser azureuser   38 Apr  1 22:54 hs37d5.fa.gz -> /mnt/1000Genome/reference/hs37d5.fa.gz
-rw-rw-r-- 1 azureuser azureuser 2813 Apr  1 22:55 hs37d5.fa.gz.fai
```

**Comment by david4096** [comment]
I was able to reproduce:

```
(server-1)➜  server-1 git:(master) ✗ python repo_dev.py add-referenceset ga4gh-example-data ../1000genomes/hs37d5.fa.gz 
Cannot index files compressed with gzip, please use bgzip
[fai_load] build FASTA index.
Cannot index files compressed with gzip, please use bgzip
[fai_load] fail to open FASTA index.
Traceback (most recent call last):
  File "repo_dev.py", line 7, in <module>
    ga4gh.cli.repo_main()
  File "/Users/david/server-1/ga4gh/cli.py", line 1672, in repo_main
    runner.run()
  File "/Users/david/server-1/ga4gh/cli.py", line 1445, in run
    self.filePath, self.moveMode, self.metadata)
  File "/Users/david/server-1/ga4gh/repo_manager.py", line 373, in addReferenceSet
    md5checksum = getReferenceChecksum(fileDestPath)
  File "/Users/david/server-1/ga4gh/repo_manager.py", line 27, in getReferenceChecksum
    inputFile = pysam.FastaFile(fastaFile)
  File "pysam/cfaidx.pyx", line 113, in pysam.cfaidx.FastaFile.__cinit__ (pysam/cfaidx.c:2025)
  File "pysam/cfaidx.pyx", line 149, in pysam.cfaidx.FastaFile._open (pysam/cfaidx.c:2845)
IOError: could not open file `ga4gh-example-data/referenceSets/hs37d5/hs37d5.fa.gz`
```

**Comment by david4096** [comment]
Pysam fails to create indexes for gzipped fasta's. First we need to `gunzip` and then `bgzip` this file. It would be nice if the repo manager would catch the pysam exception to produce a nice looking error.

**Comment by jeromekelleher** [comment]
Yes, ditto this. There doesn't seem to be any error checking on FASTA formatting in lower layers so we need to be fairly strict about what we accept as input. (For example, some versions of htslib will segfault if you provide a FASTA with no sequence ID.)

**Comment by macieksmuga** [comment]
This is related to #955 so I'm taking this up along with that issue.

**Comment by kozbo** [comment]
One added note... the bgzip that I have in my Ubuntu VM does not create an index file. So the import still failed. I tried installing the tabix tool set (sudo apt-get install tabix) but that didn't help. Not sure what I did wrong here. So I resorted to running a couple of commands in the python interpreter"
```
>>> import pysam
>>> pysam.tabix_compress("/mnt/1000Genome/reference/hs37d5.fa", "/mnt/1000Genome/reference/hs37d5.fa.gz")
>>> refFasta = pysam.FastaFile("/mnt/1000Genome/reference/hs37d5.fa.gz")
[fai_load] build FASTA index.
```
and this created the index file alongside the gz. Then the add-referenceset command worked as expected.

**Comment by diekhans** [comment]
I believe it's a two step operation:
  - bgzip
  - then index.

**Comment by jeromekelleher** [comment]
You need to call ``samtools faidx`` after bgzipping @kozbo.

**Comment by jeromekelleher** [comment]
Closed by #1166.
